concourse/pipelines/destroy-cloudfoundry.yml

```diff
@@ -47,10 +47,48 @@ resources:
       region_name: {{aws_region}}
       versioned_file: cf-secrets.yml
 
+  - name: concourse-manifest
+    type: s3-iam
+    source:
+      bucket: {{state_bucket}}
+      region_name: {{aws_region}}
+      versioned_file: concourse-manifest.yml
+
 jobs:
   - name: init
     serial: true
     plan:
+      - get: paas-cf
+      - get: concourse-manifest
+      - task: self-update-pipeline
+        config:
+          platform: linux
+          image: docker:///governmentpaas/self-update-pipelines
+          inputs:
+            - name: paas-cf
+            - name: concourse-manifest
+          params:
+            DEPLOY_ENV: {{deploy_env}}
+            BRANCH: {{branch_name}}
+            MAKEFILE_ENV_TARGET: {{makefile_env_target}}
+            SELF_UPDATE_PIPELINE: {{self_update_pipeline}}
+            PIPELINES_TO_UPDATE: {{pipeline_name}}
+          run:
+            path: bash
+            args:
+              - -e
+              - -c
+              - |
+                if [ "$SELF_UPDATE_PIPELINE" != "true" ]; then
+                  echo "Self update pipeline is disabled (set SELF_UPDATE_PIPELINE=true to enable)"
+                else
+                  echo "Self update pipeline is enabled (set SELF_UPDATE_PIPELINE=false to disable)"
+
+                  VAL_FROM_YAML=$(pwd)/paas-cf/concourse/scripts/val_from_yaml.rb
+                  export CONCOURSE_ATC_PASSWORD=$($VAL_FROM_YAML jobs.concourse.properties.atc.basic_auth_password concourse-manifest/concourse-manifest.yml)
+
+                  make -C ./paas-cf ${MAKEFILE_ENV_TARGET}
+                fi
```

Comment:
Can we move this to a script in `concourse/scripts/` to reduce duplication in each pipeline and make it easier to maintain? When we do, it should have `set -eu` at the top of it to abort on errors and unbound variables.